internal/mapping.go

```diff
@@ -142,6 +142,58 @@ func ToSpannerForeignKey(conv *Conv, srcFkName string) string {
 	return getSpannerValidName(conv, srcFkName)
 }
 
+// ToSpannerOnDelete maps the source ON DELETE action
+// to the corresponding Spanner compatible action.
+// The following mapping is followed:
+// a) CASCADE/NO ACTION -> mapped to the same as source action
+// b) all others -> NO ACTION (default)
+//
+// Since only MySQL and PostgreSQL have this functionality
+// as of yet, for other sources OnDelete fields are
+// kept empty i.e. "" is mapped to ""
+//
+// For all source actions converted to a different action,
+// an issue is appended to it's TableLevelIssues to
+// generate a warning message for the user
+func ToSpannerOnDelete(conv *Conv, srcTableId string, srcDeleteRule string) string {
+	srcDeleteRule = strings.ToUpper(srcDeleteRule)
+	if srcDeleteRule == "NO ACTION" || srcDeleteRule == "CASCADE" || srcDeleteRule == "" {
+		return srcDeleteRule
+	}
+
+	if conv.SchemaIssues == nil {
+		conv.SchemaIssues = make(map[string]TableIssues)
+	}
+	conv.SchemaIssues[srcTableId] = AddTableLevelIssue(conv.SchemaIssues[srcTableId], ForeignKeyOnDelete)
```

Comment:
This AddTableLevelIssue method seems like an overkill for what its doing. 

One can simply do: 
`conv.SchemaIssues[srcTableId].TableLevelIssues = append(conv.SchemaIssues[srcTableId].TableLevelIssues, ForeignKeyOnDelete)
`